VersionControlUI/src/DiffManager.cpp

```diff
@@ -53,73 +53,94 @@ namespace VersionControlUI
 {
 
 struct ChangeWithNodes {
-		Model::Node* oldNode_{};
-		Model::Node* newNode_{};
-		FilePersistence::ChangeType changeType_{FilePersistence::ChangeType::Unclassified};
+	Model::Node* oldNode_{};
+	Model::Node* newNode_{};
+	FilePersistence::ChangeType changeType_{FilePersistence::ChangeType::Unclassified};
+};
+
+struct DiffSetup {
+	Model::TreeManager* newVersionManager_{};
+	Model::TreeManager* oldVersionManager_{};
+	FilePersistence::GitRepository* repository_{};
 };
 
 DiffManager::DiffManager(QString oldVersion, QString newVersion, QString project,
 								 Model::SymbolMatcher contextUnitMatcher) :
 	oldVersion_{oldVersion}, newVersion_{newVersion}, project_{project}, contextUnitMatcher_{contextUnitMatcher}
 {}
 
-void DiffManager::visualize()
+DiffSetup DiffManager::initializeDiffPrerequisites()
 {
+	DiffSetup diffSetup{};
```

Comment:
Always do things as late as possible. So just before you use `diffSetup` for the first time.